be/src/storage/rowset/segment_iterator.cpp

```diff
@@ -669,7 +674,7 @@ Status SegmentIterator::_do_get_next(Chunk* result, vector<rowid_t>* rowid) {
 
     Chunk* chunk = _context->_read_chunk.get();
 
-    while ((chunk_start < chunk_capacity) & _range_iter.has_more()) {
+    while ((chunk_start < chunk_capacity) && _range_iter.has_more()) {
```

Comment:
Why change it ?

Reply:
> Why change it ?

for short-circuit computation, they have the same semantics

Reply:
you shouldn't change it.  & has better performance than &&.